src/engine/graphic/GRSystemSlice.cpp

```diff
@@ -372,6 +372,11 @@ void GRSystemSlice::Finish()
 			if (prev && !prev->getStaffState()->distanceset)
                 nextposition.y -= myrect.top;
 
+            // fixed stave spacing
+            float staffDistance = theStaff->getStaffState()->getStaffDistance();
+            if (staffDistance > 0.0)
+                nextposition.y = (theStaff->getDredgeSize() + staffDistance) * (i - 1);
```

Comment:
nextposition must be relative to the previous staff
the following code fails with this implementation:
   ```
{  
    [ a b c ],
    [	\staffFormat<distance=2> a b c ],
    [	\staffFormat<distance=3> a b c ],
    [	\staffFormat<distance=2> a b c ]
   }
```